### PR TITLE
drivers: nxp_s32_canxl: support rxfifo on new SoC version s32ze

### PR DESCRIPTION
**Pull request by haduongquang**
Add support RX FIFO on new SoC version s32ze

### CODE REVIEW

**Review by henrikbrixandersen — DISMISSED** [review]
LGTM

File: drivers/can/can_nxp_s32_canxl.c [hunk 1]
```diff
@@ -179,7 +179,7 @@ static inline int can_nxp_s32_get_rx_fifo_filter(struct can_nxp_s32_data *data)
 			continue;
 		}
 
-		if ((data->rx_fifo[0].Header.Id & mask) ==
+		if ((data->rx_fifo[queue_idx].Header.Id & mask) ==
```
Comment (manuargue) [inline drivers/can/can_nxp_s32_canxl.c:182]: is this backwards compatible with older sil cuts?

Comment (haduongquang) [reply drivers/can/can_nxp_s32_canxl.c:182]: I am afraid that, no.

Comment (manuargue) [reply drivers/can/can_nxp_s32_canxl.c:182]: if this change is breaking support for users with older Sil cuts, then IMO it should be noted in RN.
On the other hand, `zephyr/soc/nxp/s32/s32ze/Kconfig.soc` defines only `SOC_PART_NUMBER_P32Z270ADCK0MJFT`. Does the new Sil version keeps the same part numbering or are we forgetting to update this?

Comment (haduongquang) [reply drivers/can/can_nxp_s32_canxl.c:182]: I will document in RN. For the new Sil, I haven't found the name of the new cut yet, so I propose to use `SOC_PART_NUMBER_P32Z270` only. What do you think, then I'll re-push

**Review by henrikbrixandersen — DISMISSED** [review]

File: doc/releases/release-notes-4.0.rst [hunk 2]
```diff
@@ -68,6 +68,8 @@ Boards & SoC Support
 
 * Made these changes in other SoC series:
 
+  * S32Z27: Added support for new cut 2.0, the old cut (1.0, 1.1) are not fully compatibles.
```
Comment (manuargue) [inline doc/releases/release-notes-4.0.rst:71]:
```suggestion
  * NXP S32Z270: Added support for the new silicon cut version 2.0. Note that the previous versions (1.0 and 1.1) are no longer supported.
```

Comment (haduongquang) [reply doc/releases/release-notes-4.0.rst]: Thanks for your suggestion, I updated.

**Review by manuargue — APPROVED** [review]

**Review by Dat-NguyenDuy — APPROVED** [review]

**Review by bperseghetti — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by str4t0m — APPROVED** [review]

**Pull request merged**